go-controller/pkg/ovn/gateway.go

```diff
@@ -472,6 +472,7 @@ func (gw *GatewayManager) GatewayInit(
 		}
 		p := func(item *nbdb.LogicalRouterStaticRoute) bool {
 			return item.OutputPort != nil && *item.OutputPort == *lrsr.OutputPort && item.IPPrefix == lrsr.IPPrefix &&
+				// TODO: maybe this needs to be network scoped
```

Comment:
you already search via the output port, which will include the network name of the specific GR for this network. I don't see a need to add anymore search criteria